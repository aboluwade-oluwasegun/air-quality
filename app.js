import express from 'express';
import 'dotenv/config';
import bodyParser from 'body-parser';
import swaggerUi from 'swagger-ui-express';
import { airQualityRouter } from './src/routes/getAirQuality.js';
import { datetimeRouter } from './src/routes/getHighestValues.js';
import swaggerDocument from './src/utils/swaggerDoc.js';

const app = express();

app.set('trust proxy', true);

app.use((req, res, next) => {
  res.set('Access-Control-Allow-Origin', '*');
  res.set('Access-Control-Allow-Methods', 'POST');
  res.set('Access-Control-Allow-Headers', 'Content-Type, Authorization');
  if (req.method === 'OPTIONS') {
    return res.sendStatus(200);
  }
  next();
});

app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));

app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
app.use(airQualityRouter);
app.use(datetimeRouter);

app.all('*', () => {
  throw new Error('Route not found');
});

export { app };
