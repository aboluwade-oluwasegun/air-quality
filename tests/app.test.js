import supertest from 'supertest';
import { app } from '../app';

const request = supertest(app);

describe('App Tests', () => {
  test('POST /api/get-air-quality', async (done) => {
    const response = await request.post('/api/get-air-quality').send({
      lat: '35.98',
      lon: '140.33',
    });

    console.log(response.body);
    expect(response.status).toBe(200);
    expect(response.body.Result.Pollution).toMatchSnapshot({
      ts: expect.any(String),
      aqius: expect.any(Number),
      mainus: expect.any(String),
      aqicn: expect.any(Number),
      maincn: expect.any(String),
    });
  });
});
