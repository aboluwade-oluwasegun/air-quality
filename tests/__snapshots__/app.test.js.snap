// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests POST /api/get-air-quality 1`] = `
{
  "aqicn": Any<Number>,
  "aqius": Any<Number>,
  "maincn": Any<String>,
  "mainus": Any<String>,
  "ts": Any<String>,
}
`;
